include/envoy/runtime/runtime.h

```diff
@@ -83,6 +83,13 @@ class Snapshot {
   // configuration of "false" in runtime config.
   virtual bool deprecatedFeatureEnabled(const std::string& key) const PURE;
 
+  // Returns true if a runtime feature is enabled.
+  //
+  // Runtime features are used to easily allow switching between old and new code paths for high
+  // risk changes. The intent is for the old code path to be short lived - the old code path is
+  // deprecated as the feature is defaulted true, and removed with the following Envoy release.
+  virtual bool runtimeFeatureEnabled(const std::string& key) const PURE;
```

Comment:
use `absl::string_view`? It will avoid constructing std::string when caller call this with literals. but might require underlying methods take string_view.

Reply:
As written I think someone could
static std::string feature = "envoy.reloadable_features.my_feature_name"));
if (Runtime::runtimeFeatureEnabled(feature) { ...} and avoid the string conversion on hash map lookup each time.

If they do the inefficient 
if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.my_feature_name")
they're creating a string every time.

If I switch to string_view I think the map lookup will convert the string piece to the string on every single call, and there's no way to avoid it, so I think that's strictly less performant?

I could update the instructions to encourage static string on the canonical non-latch path if we think that makes more sense

Reply:
As long as it is an `absl::flat_hash_{map,set}` I think heterogenous lookup means that there is no string construction in the map lookup, even though the type in the map is `std::string`: https://abseil.io/tips/144

Reply:
TIL!   Ok, got that to work for the flat hash, but unfortunately to string_view all the way up the API we need to also change Runtime::EntryMap in include/envoy/runtime/runtime.h to have a comparator.  I'd prefer to land this as-is and do a separate PR converting to string piece, especially as none of the call sites will need to change.  What do you think of a TODO and a follow-up?

Reply:
Yeah fine to leave as TODO. `static std::string feature = "envoy.reloadable_features.my_feature_name";` is a static non-POD object initialization so there might be static initialization fiasco. string_view constructor is safe as it is constexpr.

Reply:
That sounds fine to me.